enterprise/api/v1/views/enterprise_customer_sso_configuration.py

```diff
@@ -6,19 +6,50 @@
 from rest_framework import permissions, viewsets
 from rest_framework.decorators import action
 from rest_framework.response import Response
-from rest_framework.status import HTTP_200_OK, HTTP_404_NOT_FOUND
+from rest_framework.status import HTTP_200_OK, HTTP_201_CREATED, HTTP_400_BAD_REQUEST, HTTP_404_NOT_FOUND
 
 from django.contrib import auth
+from django.core.exceptions import FieldDoesNotExist, ValidationError
+from django.db import transaction
 
 from enterprise import models
+from enterprise.api.utils import get_enterprise_customer_from_user_id
 from enterprise.api.v1 import serializers
 from enterprise.logging import getEnterpriseLogger
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerSsoConfiguration, EnterpriseCustomerUser
 from enterprise.utils import localized_utcnow
 
 User = auth.get_user_model()
 
 LOGGER = getEnterpriseLogger(__name__)
 
+BAD_CUSTOMER_ERROR = 'Must provide valid enterprise customer'
+CONFIG_UPDATE_ERROR = 'Error updating SSO configuration record'
+CONFIG_CREATE_ERROR = 'Error creating SSO configuration record'
+
+
+def check_user_part_of_customer(user, enterprise_customer):
+    """
+    Checks if a user is in an enterprise customer.
+    """
+    return enterprise_customer.enterprise_customer_users.filter(user_id=user.id).exists() or user.is_staff
+
+
+def get_customer_from_request(request):
+    """
+    Gets the enterprise customer from the request or the user.
+    """
+    if customer_uuid := request.query_params.get('enterprise_customer'):
+        return customer_uuid
+    return get_enterprise_customer_from_user_id(request.user.id)
```

Comment:
what happens if a user belongs to more than 1 enterprise customer?

Reply:
that's why I included the customer param, the requester can always simple include the enterprise customer that they want to be associated with the request. If none is supplied then all we can do is go off of the first linked one.